component-runtime-manager/src/test/java/org/talend/sdk/component/runtime/manager/service/LocalCacheServiceTest.java

```diff
@@ -15,46 +15,167 @@
-
 import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.Supplier;
 
+import org.apache.xbean.propertyeditor.PropertyEditorRegistry;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.talend.sdk.component.api.service.cache.LocalCache;
+import org.talend.sdk.component.api.service.cache.LocalCache.Element;
+import org.talend.sdk.component.api.service.configuration.LocalConfiguration;
+import org.talend.sdk.component.api.service.injector.Injector;
+import org.talend.sdk.component.runtime.manager.asm.ProxyGenerator;
+import org.talend.sdk.component.runtime.manager.reflect.ParameterModelService;
+import org.talend.sdk.component.runtime.manager.reflect.ReflectionService;
 import org.talend.sdk.component.runtime.manager.serialization.DynamicContainerFinder;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.talend.sdk.component.runtime.manager.test.Serializer.roundTrip;
+
 class LocalCacheServiceTest {
 
+
+    private boolean active;
+
+    private int  intervale;
+
+    private LocalConfiguration cacheConfig = new LocalConfiguration() {
+        @Override
+        public String get(String key) {
+            if ("test.talend.component.manager.services.cache.eviction.defaultEvictionInterval".equals(key)) {
+                return String.valueOf(intervale);
+            }
+            if ("test.talend.component.manager.services.cache.eviction.active".equals(key)) {
+                return String.valueOf(active);
+            }
+            throw new IllegalArgumentException("unknown key '" + key + "'");
+        }
+
+        @Override
+        public Set<String> keys() {
+            Set<String> keys = new HashSet<>();
+            keys.add("test.talend.component.manager.services.cache.eviction.defaultEvictionInterval");
+            keys.add("test.talend.component.manager.services.cache.eviction.active");
+            return keys;
+        }
+    };
+
+    private final LocalCacheService<String> cache = new LocalCacheService<>("LocalCacheServiceTest");
+
+
+    @BeforeEach
+    void init() {
+        this.active = false;
+        this.intervale = 0;
+        final Map<Class<?>, Object> services = new HashMap<>(2);
+       // services.put(LocalCache.class, new LocalCacheService("LocalCacheServiceTest"));
+        services.put(LocalConfiguration.class,
+                new LocalConfigurationService(Collections.singletonList(cacheConfig), "test"));
+
+
+        final PropertyEditorRegistry propertyEditorRegistry = new PropertyEditorRegistry();
```

Comment:
not sure this helps the test, at the end it just gets back the instance you created just before

Reply:
I just followed java API to inject value to services. If you suggest better, i can do it.

Reply:
but you don't test the injection, you test the instance you injected so, until I misunderstood the test, it does not do anything for this particular test :thinking: